### PR TITLE
[FLINK-2851] Move PythonAPI to flink-libraries

### PR DESCRIPTION
**Pull request by zentol**

### CODE REVIEW

File: flink-libraries/flink-python/src/main/java/org/apache/flink/python/api/PythonOperationInfo.java [hunk 1]
```diff
@@ -1,3 +1,5 @@
+package org.apache.flink.python.api;
```
Comment (sachingoel0101) [inline flink-libraries/flink-python/src/main/java/org/apache/flink/python/api/PythonOperationInfo.java]: This should be after the license.

Comment (zentol) [inline flink-libraries/flink-python/src/main/java/org/apache/flink/python/api/PythonOperationInfo.java]: Fixed.

**Comment by mxm** [comment]
Looks like you did quite some refactoring e.g. in `FlinkPlanBinder`. Could you elaborate on your changes besides moving everything to flink-libraries?

Otherwise +1 to merge.

**Comment by zentol** [comment]
I moved the PythonOperationInfo into a separate file (along with a few static helper methods). Apart from that i only merged the PlanBinder and PythonPlanBinder classes and simplified the code where it made sense.

There is no longer a differentiation between Operation (non-UDF, handled by generic code) and AbstractOperations(UDF, handled by language-specific code), because there is no longer a reason to do that.

Previously, to create e.g a flatMap operation you would first execute this code in PlanBinder:
```
private void createFlatMapOperation(INFO info) {
	DataSet op1 = (DataSet) sets.get(info.parentID);
	sets.put(info.setID, applyFlatMapOperation(op1, info));}
```
and then in PythonPlanBinder:
```
@Override
protected DataSet applyFlatMapOperation(DataSet op1, PythonOperationInfo info) {
	return op1.mapPartition(new PythonMapPartition(info.setID, info.types)).name(info.name);}
```
these methods introduced a lot of clutter so i merged them.

I can't think of anything else i changed.

**Comment by mxm** [comment]
Ah, that makes sense and looks much cleaner. Thank you. +1 to merge.

**Comment by fhueske** [comment]
Hi @zentol, this is another step towards the Maven module refactoring that we decided to do.
I think you can merge this PR to the master branch.

**Pull request closed without merging**